test/tests/functional/pbs_suspend_resume_accounting.py

```diff
@@ -0,0 +1,356 @@
+
+
+class TestSuspendResumeAccounting(TestFunctional):
+    """
+    Testsuite for verifying accounting of
+    suspend/resume events of job
+    """
+
+    script = ['#!/bin/bash\nfor ((c=1; c <= 1000000000; c++));']
+    script += ['do']
+    script += ['sleep 1']
+    script += ['done']
+
+    def test_suspend_resume_job_signal(self):
+        """
+        Test case to verify accounting suspend
+        and resume records when the events are
+        triggered by client for one job.
+        """
+        j = Job(TEST_USER)
+        j.create_script(self.script)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid)
+
+        self.server.sigjob(jobid=jid, signal="suspend")
+
+        record = 'z;%s;requestor=%s@%s .*resources_used.' % \
+                 (jid, self.server.current_user, self.server.hostname)
+        self.server.accounting_match(msg=record, id=jid, regexp=True)
+
+        self.server.sigjob(jobid=jid, signal="resume")
+        record = 'r;%s;requestor=Scheduler@%s' % \
+                 (jid, self.server.hostname)
+        self.server.accounting_match(msg=record, id=jid)
+
+    @skipOnCpuSet
+    def test_suspend_resume_job_array_signal(self):
+        """
+        Test case to verify accounting suspend
+        and resume records when the events are
+        triggered by client for a job array.
+        """
+        a = {ATTR_rescavail + '.ncpus': 2}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+
+        j = Job(TEST_USER)
+        j.create_script(self.script)
+        j.set_attributes({ATTR_J: '1-2'})
+        jid = self.server.submit(j)
+
+        subjobs = self.server.status(JOB, id=jid, extend='t')
+        sub_jid1 = subjobs[1]['id']
+        sub_jid2 = subjobs[2]['id']
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=sub_jid1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=sub_jid2)
+
+        # suspend job
+        self.server.sigjob(jobid=jid, signal="suspend")
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=sub_jid1)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=sub_jid2)
+
+        record = 'z;%s;requestor=%s@%s resources_used.' % \
+                 (sub_jid1, self.server.current_user, self.server.hostname)
+        self.server.accounting_match(msg=record, id=sub_jid1)
+
+        record = 'z;%s;requestor=%s@%s resources_used.' % \
+                 (sub_jid2, self.server.current_user, self.server.hostname)
+        self.server.accounting_match(msg=record, id=sub_jid2)
+
+        self.server.sigjob(jobid=jid, signal="resume")
+        record = 'r;%s;requestor=Scheduler@%s' % \
+                 (sub_jid1, self.server.hostname)
+        self.server.accounting_match(msg=record, id=sub_jid1)
+
+        record = 'r;%s;requestor=Scheduler@%s' % \
+                 (sub_jid2, self.server.hostname)
+        self.server.accounting_match(msg=record, id=sub_jid2)
+
+    def test_interactive_job_suspend_resume(self):
```

Comment:
Why does it matter if a job is interactive or batch for suspend resume?

Reply:
I was actually verifying for all the scenarios of job suspend. so added for interactive job as well. If it's not really necessary, I can remove it. Please let me know.

Reply:
We could argue the same for job arrays as well right? I think for the sake of completeness, it's good to keep this test as well.